src/go/guestagent/pkg/procnet/scanner_linux.go

```diff
@@ -0,0 +1,239 @@
+				continue
+			}
+			newPortMap := make(nat.PortMap)
+			for _, entry := range entries {
+				if err := addValidProtoEntryToPortMap(entry, newPortMap); err != nil {
+					log.Errorf("failed to create portMapping for entry: %w", err)
+					continue
+				}
+			}
+
+			// Add new ports
+			for port, bindings := range newPortMap {
+				if _, exists := previousPortMap[port]; !exists {
+					log.Infof("/proc/net scanner added port: %s -> %+v", port, bindings)
+					err := p.tracker.Add(generateID(fmt.Sprintf("%s/%s", port.Proto(), port.Port())), nat.PortMap{
+						port: bindings,
+					})
+					if err != nil {
+						log.Errorf("/proc/net scanner failed to add port: %s", err)
+						continue
+					}
+					if err = p.execLoopbackIPtablesRule(bindings, port, Append); err != nil {
+						log.Errorf("/proc/net scanner creating loopback iptable rules for portbinding: %v failed: %s", bindings, err)
+					}
+				}
+			}
+
+			// Remove old ports
+			for port, previousBindings := range previousPortMap {
+				if _, exists := newPortMap[port]; !exists {
+					log.Infof("/proc/net scanner removed port: %s -> %+v", port, previousBindings)
+					err := p.tracker.Remove(generateID(fmt.Sprintf("%s/%s", port.Proto(), port.Port())))
+					if err != nil {
+						log.Errorf("/proc/net scanner failed to remove port: %s", err)
+						continue
+					}
+
+					if err = p.execLoopbackIPtablesRule(previousBindings, port, Delete); err != nil {
+						log.Errorf("/proc/net scanner deleting loopback iptable rules for portbinding: %v failed: %s", previousBindings, err)
+					}
+				}
+			}
+
+			previousPortMap = newPortMap
+		}
+	}
+}
+
+// execLoopbackIPtablesRule modifies iptables NAT rules to handle loopback traffic for a specified port
+// and protocol. This function is only necessary when the container is using the host network driver
+// (i.e., with --network=host), as in this case the container shares the host's network namespace.
+//
+// When using the host network driver, the network traffic that is bound to 127.0.0.1 would need to be
+// redirected from outside of the network namespace to the localhost (127.0.0.1). This function adds or
+// removes DNAT rules that allow traffic to be forwarded to the specified port on localhost, based on
+// the provided 'action' ('append' or 'delete').
+//
+// The function iterates over the provided list of port bindings and, for each binding where the HostIP
+// is set to 127.0.0.1, it constructs and executes the corresponding iptables command to either add or
+// delete the appropriate DNAT rule.
+//
+// The iptables rule ensures that incoming traffic from outside  of the network namespace (i.e.,
+// from the host windows machine) on the specified port and protocol is redirected to the same
+// port on localhost, where the container's service can be accessed.
+//
+// Example iptables rule when 'action' is "append":
+// iptables -t nat -A PREROUTING -p tcp --dport 8009 -j DNAT --to-destination 127.0.0.1:8009
```

Comment:
`go doc` doesn't respect line wrapping inside a paragraph; if you want that to look like code you need [indents](https://go.dev/doc/comment#code).